src/Core/AbstractApi.php

```diff
@@ -91,16 +95,30 @@ public function __construct($configuration = [], ?CredentialProvider $credential
     abstract protected function getSignatureScopeName(): string;
 
     /**
-     * @param string[]|string[][]      $headers headers names provided as keys or as part of values
-     * @param string|resource|callable $body
+     * @param string[]|string[][]                    $headers headers names provided as keys or as part of values
+     * @param string|resource|callable|iterable|null $body
      */
     final protected function getResponse(string $method, $body, $headers = [], ?string $endpoint = null): ResponseInterface
     {
         if (!isset($headers['content-type'])) {
             $headers['content-type'] = 'text/plain';
         }
 
-        $request = new Request($method, $this->fillEndpoint($endpoint), $headers, $body);
+        $stream = null;
+        if (null === $body || \is_string($body)) {
+            $stream = new StringStream($body ?? '');
+        } elseif (\is_resource($body)) {
+            $stream = new ResourceStream($body);
+        } elseif (\is_callable($body)) {
+            $stream = new CallableStream($body);
+        } elseif (\is_iterable($body)) {
+            $stream = new IterableStream($body);
+        }
```

Comment:
If we decide to not make streams internal, we should move this to a `StreamFactory`

Reply:
Added a streamFactory. It's simpler